pymeasure/instruments/agilent/agilent8722ES.py

```diff
@@ -121,48 +131,51 @@ def set_IF_bandwidth(self, bandwidth):
                                  "8722ES exceeded")
 
     def set_averaging(self, averages):
-        """ Turns on averaging of a specific number between 0 and 999
-        """
-        if int(averages) > 999 or int(averages) < 0:
-            raise RangeException("Averaging must be in the range 0 to 999")
+        """Sets the number of averages and enables/disables averaging"""
+        averages = int(averages)
+        if not 1 <= averages <= 999:
+            assert RangeException("Invalid number of averages -", averages)
+        self.averages = averages
+        if averages > 1:
+            self.enable_averaging()
         else:
-            self.write("AVERO1")
-            self.write("AVERFACT%d" % int(averages))
+            self.disable_averaging()
 
     def disable_averaging(self):
```

Comment:
You could consider making a `Instrument.control` for this, such that you could use `Agilent8722ES.averaging = True`, and have `disable_averaging` and `enable_averaging` as a wrapper for this control.

Reply:
I agree!